src/ngx_http_lua_headers_in.c

```diff
@@ -645,6 +645,78 @@ ngx_http_clear_builtin_header(ngx_http_request_t *r,
 }
 
 
+static    ngx_hash_init_t             lua_req_header_hash;
+static    u_char header_string[512];
+ngx_int_t
+lua_req_set_header_hash_init(void)
+{
+    ngx_int_t                   rc;
+    ngx_uint_t                  i;
+    ngx_uint_t                  pos = 0;
+    ngx_uint_t                  key_max_len = 0;
+    ngx_str_t                   tmp_key = {0, &header_string[0]};
+    ngx_hash_keys_arrays_t      ha;
+    ngx_http_lua_set_header_t        *handlers = ngx_http_lua_set_handlers;
+
+    ngx_memzero(&ha, sizeof(ngx_hash_keys_arrays_t));
+
+    ha.temp_pool = ngx_create_pool(NGX_DEFAULT_POOL_SIZE, ngx_cycle->log);
+    if (ha.temp_pool == NULL) {
+        return NGX_ERROR;
+    }
+
+    ha.pool = ngx_create_pool(NGX_DEFAULT_POOL_SIZE, ngx_cycle->log);
+    if (ha.pool == NULL) {
+        goto failed;
+    }
+
+    if (ngx_hash_keys_array_init(&ha, 50) != NGX_OK) {
+        goto failed;
+    }
+
+    for (i = 0; handlers[i].name.len; i++) {
+        tmp_key.len = handlers[i].name.len;
+        if (tmp_key.len > key_max_len) {
+            key_max_len = tmp_key.len;
+        }
+        if (pos + tmp_key.len + 1 >= sizeof(header_string)) {
+            goto failed;
+        }
+        tmp_key.data = &header_string[pos];
+        strcpy((char *)tmp_key.data, (char *)handlers[i].name.data);
+        pos += tmp_key.len + 1;
+
+        rc = ngx_hash_add_key(&ha, &tmp_key, (void *)(i+1), 0);
+        dd("add key=%.*s len=%d idx=%d", (int)tmp_key.len, tmp_key.data, (int)tmp_key.len, (int)i);
+        if (rc == NGX_ERROR) {
+            return NGX_ERROR;
+        }
+    }
+
+    lua_req_header_hash.key = ngx_hash_key_lc;
+    lua_req_header_hash.max_size = 512;
+    lua_req_header_hash.bucket_size = ngx_align((key_max_len + 3 * sizeof(void *)), sizeof(void *));
+    lua_req_header_hash.name = "req_set_header_hash";
+    lua_req_header_hash.pool = ha.pool;
+
+    if (ha.keys.nelts) {
+        lua_req_header_hash.hash = NULL;
+        lua_req_header_hash.temp_pool = NULL;
+
+        if (ngx_hash_init(&lua_req_header_hash, ha.keys.elts, ha.keys.nelts) != NGX_OK) {
+            goto failed;
+        }
+    }
+
+    return NGX_OK;
+failed:
```

Comment:
Style: need blank lines around the statement labels.

BTW, the `ngx-releng` tool from the nginx-devel-utils repo can catch such coding style issues. Please run this tool against your changes. Thank you.